### PR TITLE
fix linting with mypy + precommit?

### PR DESCRIPTION
**Pull request by logan-markewich**
The current pre-commit config was missing errors in mypy (I purposely tried to make typing errors, and they weren't detected)

From my local testing, this seems to have fixed the issue (and also identified some issues in core!)

### CODE REVIEW

File: llama-index-core/llama_index/core/settings.py [hunk 1]
```diff
@@ -113,11 +113,11 @@ def tokenizer(self) -> Callable[[str], List[Any]]:
         """Get the tokenizer."""
         import llama_index.core
 
-        if llama_index.global_tokenizer is None:
+        if llama_index.core.global_tokenizer is None:
```
Comment (nerdai) [inline llama-index-core/llama_index/core/settings.py:116]: I could've sworn we've fixed this before. I remember doing search for global_tokenizer specifically. Maybe got overwritten?

**Review by nerdai — APPROVED** [review]

File: llama-index-core/llama_index/core/settings.py (hunk 1 above)
Comment (logan-markewich) [reply llama-index-core/llama_index/core/settings.py:116]: it might have gotten overwritten at some point lol

Comment (logan-markewich) [reply llama-index-core/llama_index/core/settings.py:116]: lots of moving parts lately

Comment (nerdai) [reply llama-index-core/llama_index/core/settings.py:116]: Lol yah that's for sure. Oh well -- we do our best!

**Pull request merged**